### PR TITLE
plumb actions and redux store for new conflict protocol changes

### PR DESCRIPTION
**Pull request by songgao**
@jakob223 I created a temporary branch by rebasing `jakob/KBFS-4151-resolve-manually-link` to master, and am opening this PR against that branch so it's clear what's changed. It's mostly plumbing, plus fixing flow errors by changing as less as possible. Let me know what you think!

### CODE REVIEW

File: shared/actions/fs/index.js [hunk 1]
```diff
@@ -967,12 +966,14 @@ const waitForKbfsDaemon = (state, action) => {
...
     path: Constants.pathToRPCPath(action.payload.tlfPath),
-  }).then(() =>
-    FsGen.createTlfCrStatusChanged({
-      status: 'in-manual-resolution',
-      tlfPath: action.payload.tlfPath,
-    })
-  ) // TODO: deal with errors
+  }).then(() => FsGen.createFavoritesLoad()) // TODO: deal with errors
+// songgao: perhaps if this errors we should just let it black bar? I guess
+// depends on if/how we expect this to fail.
```
Comment (jakob223) [inline shared/actions/fs/index.js]: Only situation I can that maybe shouldn't black bar is if the user is running multiple KBFSes and thus has a nil CR DB. I think black bar is probably fine though.

File: shared/constants/fs.js [hunk 2]
```diff
@@ -99,13 +99,33 @@ export const makeTlfSyncPartial: I.RecordFactory<Types._TlfSyncPartial> = I.Reco
...
+export const conflictStateAutomaticResolvingStuck = makeConflictStateAutomaticResolving({isStuck: true})
+
+export const makeConflictStateManualResolvingServerView: I.RecordFactory<Types._ConflictStateManualResolvingServerView> = I.Record(
+  {
+    localViews: I.List(),
+    type: 'manual-server-view',
+  }
+)
+
+export const makeConflictStateManualResolvingLocalView: I.RecordFactory<Types._ConflictStateManualResolvingLocalView> = I.Record(
```
Comment (jakob223) [inline shared/constants/fs.js]: are we going to have a separate `tlfs` entry for each of the views?

File: shared/fs/banner/conflict-banner.js [hunk 3]
```diff
@@ -69,26 +65,26 @@ const ConflictBanner = (props: Props) => {
   const onSeeOtherView = props.isUnmergedView ? onSeeServerView : onSeeLocalView
 
   let actions = []
-  switch (props.conflictState) {
-    case 'in-conflict-stuck':
-      actions = [startRes, feedbackAction, helpAction]
+  switch (props.conflictState.type) {
+    case 'automatic':
+      actions = props.conflictState.isStuck
```
Comment (jakob223) [inline shared/fs/banner/conflict-banner.js]: These are the same - maybe just `actions = [...]`?

File: shared/reducers/__tests__/fs.js [hunk 4]
```diff
@@ -184,12 +184,19 @@ describe('fs reducer', () => {
 
   test('favorritesLoaded: reuse tlf', () => {
```
Comment (jakob223) [inline shared/reducers/__tests__/fs.js]: whoa I didn't know about these tests

File: shared/reducers/fs.js [hunk 5]
```diff
@@ -116,42 +116,32 @@ const updatePathItem = (
...
+const updateTlf = (oldTlf?: ?Types.Tlf, newTlf: Types.Tlf): Types.Tlf =>
+  oldTlf
+    ? oldTlf.merge(
+        newTlf.withMutations(n =>
+          n
+            .set(
+              'syncConfig',
+              // This comes from a different RPC for now. So make sure we don't
+              // override non-nil syncConfig with a nil one.
+              !newTlf.syncConfig || I.is(newTlf.syncConfig, oldTlf.syncConfig)
```
Comment (jakob223) [inline shared/reducers/fs.js]: Can you put parentheses around this line to make order of operations clearer? I assumed at first that the `||` was being an "if this is not null then that" thing.

File: shared/constants/fs.js (hunk 2 above)
Comment (songgao) [reply shared/constants/fs.js]: Yeah, when we split, we’ll have one for server view and one for local view

File: shared/reducers/__tests__/fs.js (hunk 4 above)
Comment (songgao) [reply shared/reducers/__tests__/fs.js]: Yep! For Fs we only have them for the reducer though as they tend to be complex and error prone.

File: shared/reducers/fs.js (hunk 5 above)
Comment (songgao) [reply shared/reducers/fs.js]: I think my prettier/linter remove them usually, but I’ll try and see if I am make it clearer

**Comment by jzila** [comment]
@songgao are we able to merge this?

**Comment by jakob223** [comment]
I think this is waiting on https://github.com/keybase/client/pull/17491

**Comment by songgao** [comment]
This ends up impossible to rebase using `git rebase` or a edited patch file. I'll close this for now and open a new one with man-powered super rebase

**Comment by jakob223** [comment]
Oh no, sorry that happened :|

**Pull request closed without merging**